### PR TITLE
Pin pint to >=0.22

### PR DESCRIPTION
**Pull request by dcherian**
- [x] Closes https://github.com/pydata/xarray/issues/7971
- [x] Closes https://github.com/pydata/xarray/issues/8437.

We were previously pinned to `<0.21`
Removing the pin didn't change the env but with `>=0.21` we get `0.22` which works.

### CODE REVIEW

**Comment by keewis** [comment]
That's weird, is there another library that pins `pint`?

If we merge this it should close #7971 in addition to #8437.

**Comment by dcherian** [comment]
I bumped the min-deps version too, seems like many things have been fixed since 0.19

File: ci/requirements/min-all-deps.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ dependencies:
   - numpy=1.22
   - packaging=21.3
   - pandas=1.4
-  - pint=0.19
+  - pint>=0.22
```
Comment (keewis) [inline ci/requirements/min-all-deps.yml:38]:
```suggestion
  - pint=0.22
```

**Review by keewis — APPROVED** [review]
`mypy` looks unrelated

**Comment by max-sixty** [comment]
I'll hit the button if that's OK!

**Pull request merged**